src/collection.js

```diff
@@ -667,6 +656,73 @@ export default class Collection {
   }
 
   /**
+   * Imports the responses of pushed changes into the local database.
+   * Basically it stores the timestamp assigned by the server into the local
+   * database.
+   * @param  {SyncResultObject} syncResultObject The sync result object.
+   * @param  {Array}            toApplyLocally   The list of changes to import in the local database.
+   * @param  {String}           strategy         The {@link Collection.strategy}.
+   * @return {Promise}
+   */
+  async importPublications(syncResultObject, toApplyLocally, strategy) {
+    const toDeleteLocally = toApplyLocally.filter((r) => r.deleted);
+    const toUpdateLocally = toApplyLocally.filter((r) => !r.deleted);
+
+    const conflicts = syncResultObject.conflicts;
+
+    const {published, resolved} = await this.db.execute((transaction) => {
+      const updated = toUpdateLocally.map((record) => {
+        const synced = markSynced(record);
+        transaction.update(synced);
+        return synced;
+      });
+      const deleted = toDeleteLocally.map((record) => {
+        transaction.delete(record.id);
+        // Amend result data with the deleted attribute set
+        return {id: record.id, deleted: true};
+      });
+      const published = updated.concat(deleted);
+      // Handle conflicts, if any
+      const resolved = this._handleConflicts(transaction, conflicts, strategy);
```

Comment:
It seems a little strange to have `importPublications`, which I assumed was just going to handle updating state about records that were successfully published, to also handle conflicts (records that weren't actually published). Can we have the caller manage the transaction and call both `importPublications` and `_handleConflicts`? Or maybe we can change the function name to something like `importPushResult`, and make `conflicts` be a parameter to the function?